core/session.py

```diff
@@ -199,56 +192,27 @@ def reset(self) -> None:
 
             :warning: Everything is flushed.
         """
-        if self.cookieKey:
-            db.Delete(db.Key(self.kindName, self.cookieKey))
+        if self.cookie_key:
+            db.Delete(db.Key(self.kindName, self.cookie_key))
+            from viur.core import securitykey
+            securitykey.clear_session_skeys(self.cookie_key)
 
-        self.cookieKey = utils.generateRandomString(42)
-        self.staticSecurityKey = utils.generateRandomString(13)
-        self.securityKey = utils.generateRandomString(13)
+        self.cookie_key = utils.generateRandomString(42)
+        self.static_security_key = utils.generateRandomString(13)
+        self.session.clear()
```

Comment:
Since `viur.datastore.Entity` does not have a custom `clear` method, I would also recommend doing to reset the key to make sure the new session gets a new key. Maybe there are some tasks or instances that hold references to the old session. Then I think (I haven't invested time in researching this) it might be possible to take over the new session incorrectly.

```py
self.session.key = None
```

Reply:
Well, the `session` property of the `Session` class is awkward, as it is a `db.Entity` which is only held in memory to change session properties and which is written into the `data` property of a `db.Entity` named `dbSession` in `Session.save()`. Therefore the key of `session.key` is never set to anything, and therefore is not required to be set to `None`. I don't know why this was solved this way, maybe it's only bad code that simply can be replaced by a dict.